core/models.py

```diff
@@ -164,40 +161,142 @@ def __str__(self):
 class ClassificationModel(BaseModel):
     name = models.CharField(max_length=200)
     version = models.CharField(max_length=20)
-    url = models.TextField(null=True, blank=True)
-    description = models.TextField()
+    model_uri = models.TextField(null=True, blank=True)
+    defaults = models.JSONField(default=dict)
+
+    class Meta:
+        unique_together = ['name', 'version', 'model_uri']
 
     def __str__(self):
-        return self.name
+        return str(self.id)
 
 
 class ClassificationPredictions(BaseModel):
     entry = models.OneToOneField(Entry, on_delete=models.CASCADE)
     project = models.ForeignKey(Project, on_delete=models.CASCADE)
     model = models.ForeignKey(ClassificationModel, on_delete=models.CASCADE)
-    embeddings = models.JSONField()
-    """
-    NOTE: All predictions will be of the following form.
-    IMPORTANT!! These may change over time. Need to migrate accordingly in the future.
-    {
-        "subpillars_1d": ["XXX"],
-        "sectors": ["XXX"],
-        "subpillars_2d": ["XXX"],
-        "age": ["XXX"],
-        "gender": ["XXX"],
-        "affected_groups": ["XXX"],
-        "specific_needs_groups": ["XXX"]
-        "severity": ["XXX"],
-    }
-    """
-    original_en_predictions = models.JSONField(default=dict)
-    original_es_predictions = models.JSONField(default=dict)
-    original_fr_predictions = models.JSONField(default=dict)
-    original_pt_predictions = models.JSONField(default=dict)
-    nlp_en_predictions = models.JSONField(default=dict)
-    nlp_es_predictions = models.JSONField(default=dict)
-    nlp_fr_predictions = models.JSONField(default=dict)
-    nlp_pt_predictions = models.JSONField(default=dict)
+    embeddings = ArrayField(models.FloatField(blank=True), blank=True, null=True)
+    subpillars_1d = ArrayField(
+        models.CharField(max_length=100, blank=True),
+        blank=True,
+        null=True
+    )
+    sectors = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    subpillars_2d = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    age = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    gender = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    affected_groups = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    specific_needs_groups = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+    severity = ArrayField(
+        models.CharField(max_length=100, blank=True), blank=True, null=True
+    )
+
+    def __str__(self):
+        return str(self.entry.original_entry_id)
+
+
+class ProjectWisePerfMetrics(models.Model):
+    project_id = models.PositiveIntegerField()
```

Comment:
Is this the project id from deep or the project object in the nlp server. Please add a comment as this is quite confusing.

Reply:
@bewakes this is project id from deep. i will add a comment for this.